app/models/item.rb

```diff
@@ -3,7 +3,15 @@ class Item < ActiveRecord::Base
 
   validates_presence_of :name, :price, :quantity, :shelf_life_days
 
-  def expires
-    Date.today + shelf_life_days.days
+  def expires_on
+    created_at + shelf_life_days.days
+  end
+
+  def self.valid_items
+    Item.all.select { |item| item.expires_on >= Date.today }
+  end
```

Comment:
This would be better as a scope but it isn't a query. You are pulling every `Item` in the database and then performing an enumeration `.select` on them.

Consider using an active-record query and moving this to a scope as @dkniffin has suggested